### PR TITLE
Adds requestBodies and other missing fields to the components object.

### LINKED ISSUE #1160: Missing components for `refPostBody` and `refPutBody` references
**Describe the bug**
`POST */$ref` and `PUT */$ref` operations are missing `refPostBody` and `refPutBody` components.

``` yml
...
paths:
  '/applications/{application-id}/owners/$ref':
    post:
      tags:
        - applications.directoryObject
      summary: Create new navigation property ref to owners for applications
      operationId: applications_CreateOwnersByRef
      parameters:
        - name: application-id
          in: path
          description: 'key: id of application'
          required: true
          schema:
            type: string
          x-ms-docs-key-type: application
      requestBody:
        $ref: '#/components/requestBodies/refPostBody' # is missing
      responses:
        '204':
          description: Success
        default:
          $ref: '#/components/responses/error'
      x-ms-docs-operation-type: operation
components:
  responses:
    error:
      description: error
security:
  - azureaadv2: [ ]
```

**To Reproduce**
1. Go to https://graphexplorerapi.azurewebsites.net/$openapi?url=/applications/{application-id}/owners/$ref&openapiversion=3&style=Powershell&graphVersion=v1.0
2. Inspect its POST operation.
3. You'll see the specified reference model definition of the request body is missing in the OpenAPI components object.

**Expected behavior**
Request bodies specified in a reference should be present in the components.
``` diff
...
paths:
  '/applications/{application-id}/owners/$ref':
    post:
      tags:
        - applications.directoryObject
      summary: Create new navigation property ref to owners for applications
      operationId: applications_CreateOwnersByRef
      parameters:
        - name: application-id
          in: path
          description: 'key: id of application'
          required: true
          schema:
            type: string
          x-ms-docs-key-type: application
      requestBody:
        $ref: '#/components/requestBodies/refPostBody'
      responses:
        '204':
          description: Success
        default:
          $ref: '#/components/responses/error'
      x-ms-docs-operation-type: operation
components:
  responses:
    error:
      description: error
  requestBodies:
+    refPostBody:
+      description: New navigation property ref value
+      content:
+        application/json:
+          schema:
+            $ref: '#/components/schemas/ReferenceCreate'
+      required: true
+    refPutBody:
+      description: New navigation property ref values
+      content:
+        application/json:
+          schema:
+            $ref: '#/components/schemas/ReferenceUpdate'
+      required: true
+  schemas:
+    ReferenceUpdate:
+      type: object
+      properties:
+        '@odata.id':
+          type: string
+        '@odata.type':
+          type: string
+          nullable: true
+    ReferenceCreate:
+      type: object
+      properties:
+        '@odata.id':
+          type: string
+      additionalProperties:
+        type: object
security:
  - azureaadv2: [ ]
```

### PR DESCRIPTION
**Pull request by peombwa**
## Overview

This PR fixes #1160 by:
- Ensuring we copy references of all `Components` object fields to the sliced OpenAPI document. See https://swagger.io/specification/#components-object.
- Updates unit tests to reflect the change above.

### CODE REVIEW

**Review by MaggieKimani1 — APPROVED** [review]

File: OpenAPIService/OpenApiService.cs [hunk 1]
```diff
@@ -803,6 +821,19 @@ private static bool AddReferences(OpenApiComponents newComponents, OpenApiCompon
...
+            foreach (var item in newComponents.RequestBodies)
+            {
+                if (!target.RequestBodies.ContainsKey(item.Key))
+                {
+                    moreStuff = true;
+                    target.RequestBodies.Add(item);
+                }
+            }
+            return moreStuff;
+        }
```
Comment (baywet) [inline OpenAPIService/OpenApiService.cs:835]:
```suggestion
        private static bool AddComponentsRequestBodies(OpenApiComponents newComponents, OpenApiComponents target)
        {
            var bodiesToAdd = newComponents.RequestBodies.Where(x => !target.RequestBodies.ContainsKey(x.Key));
            foreach (var item in bodiesToAdd)
            {
                target.RequestBodies.Add(item);
            }
            return bodiesToAdd.Any();
        }
```

File: OpenAPIService/OpenApiService.cs [hunk 2]
```diff
@@ -774,6 +774,15 @@ private static void CopyReferences(OpenApiDocument target)
         private static bool AddReferences(OpenApiComponents newComponents, OpenApiComponents target)
         {
             var moreStuff = false;
+            moreStuff = AddComponentsSchemas(newComponents, target, moreStuff);
+            moreStuff = AddComponentsParameters(newComponents, target, moreStuff);
+            moreStuff = AddComponentsResponses(newComponents, target, moreStuff);
+            moreStuff = AddComponentsRequestBodies(newComponents, target, moreStuff);
```
Comment (baywet) [inline OpenAPIService/OpenApiService.cs:780]:
```suggestion
            moreStuff |= AddComponentsRequestBodies(newComponents, target);
```

Comment (baywet) [inline OpenAPIService/OpenApiService.cs:780]:
```suggestion
            moreStuff |= AddComponentsSchemas(newComponents, target, moreStuff);
            moreStuff |= AddComponentsParameters(newComponents, target, moreStuff);
            moreStuff |= AddComponentsResponses(newComponents, target, moreStuff);
            moreStuff |= AddComponentsRequestBodies(newComponents, target, moreStuff);
```

(I'm not sure resetting the more stuff switch after each method was what was desired)

**Review by baywet — CHANGES_REQUESTED** [review]

File: OpenAPIService/OpenApiService.cs (hunk 1 above)
Comment (baywet) [reply OpenAPIService/OpenApiService.cs]: ah you didn't like my Linq expression? 🤣

**Review by baywet — APPROVED** [review]

**Pull request merged**